test/snippet/io/sequence_file/sequence_file_input.cpp

```diff
@@ -0,0 +1,222 @@
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/alphabet/quality/all.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/container/bitcompressed_vector.hpp>
+#include <seqan3/range/view/get.hpp>
+#include <seqan3/std/view/filter.hpp>
+
+using namespace seqan3;
+
+auto tmp_dir = std::filesystem::temp_directory_path();
+
+//! [data_storage]
+struct data_storage_t
+{
+    concatenated_sequences<dna5_vector>  sequences;
+    concatenated_sequences<std::string>  ids;
+};
+
+data_storage_t data_storage; // a global or globally used variable in your program
+//! [data_storage]
+
+// Todo: change std::vector to bitcompressed_vector once #764 is fixed
+//! [trait_overwrite]
+struct my_traits : sequence_file_input_default_traits_dna
+{
+    using sequence_alphabet = dna4;               // instead of dna5
+
+    template <typename alph>
+    using sequence_container = std::vector<alph>; // must be defined as a template!
+};
+
+// [...] main
+
+    sequence_file_input<my_traits> fin{tmp_dir/"my.fasta"};
+
+// [...]
+//! [trait_overwrite]
+
+int main()
+{
+{
+// Create a /tmp/my.fasta file.
+sequence_file_output fout{tmp_dir/"my.fasta"};
+fout.emplace_back("ACGT"_dna4, "TEST1");
+fout.emplace_back("AGGCTGA"_dna4, "Test2");
+fout.emplace_back("GGAGTATAATATATATATATATAT"_dna4, "Test3");
+}
+
+{
+// Create a /tmp/my_seq_qual.fasta file.
+sequence_file_output fout{tmp_dir/"my.fastq", fields<field::ID, field::SEQ_QUAL>{}};
+std::vector<qualified<dna4, phred42>> my_qualified{qualified<dna4, phred42>{'A'_dna4, phred42{7}},
+                                                   qualified<dna4, phred42>{'C'_dna4, phred42{6}},
+                                                   qualified<dna4, phred42>{'G'_dna4, phred42{5}},
+                                                   qualified<dna4, phred42>{'T'_dna4, phred42{4}}};
```

Comment:
wouldn't this also work (an be shorter):
```cpp
std::vector<qualified<dna4, phred42>> my_qualified{{'A'_dna4, phred42{7}},
                                                   {'C'_dna4, phred42{6}},
                                                   {'G'_dna4, phred42{5}},
                                                   {'T'_dna4, phred42{4}}};
```

Also the phreds have literals now ☝️